test/testdrive/kafka-stats.td

```diff
@@ -0,0 +1,92 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test the creation and removal of entries in mz_kafka_consumer_statistics
+
+$ set schema={
+    "type": "record",
+    "name": "row",
+    "fields": [
+      {"name": "a", "type": "long"},
+      {"name": "b", "type": "long"}
+    ]
+  }
+
+$ kafka-create-topic topic=data
+
+> CREATE SOURCE data
+  FROM KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-data-${testdrive.seed}'
+  WITH (statistics_interval_ms = 1000)
+  FORMAT AVRO USING SCHEMA '${schema}'
+
+> CREATE MATERIALIZED VIEW test1 AS
+  SELECT b, sum(a) FROM data GROUP BY b
+
+> SELECT * FROM test1
+b  sum
+------
+
+$ kafka-ingest format=avro topic=data schema=${schema} timestamp=1
+{"a": 1, "b": 1}
+{"a": 2, "b": 1}
+{"a": 3, "b": 1}
+{"a": 1, "b": 2}
+
+> SELECT * FROM test1
+b  sum
+------
+1  6
+2  1
+
+# There should only be metrics from a single partition consumer
+> SELECT count(*) FROM mz_kafka_consumer_statistics
```

Comment:
is there any way to have a test that actually verifies all the columns look correct?

Reply:
The main issue is that the ``rxbytes` field depends on the size of the Avro encoding. Also, the offset stuff might be broker version dependent.

Reply:
Happy to assert on the results; I'm concerned they might be brittle.

Reply:
I think having a test that includes as many columns as you feel confident about would be fine, just to be sure that the things that it captures that are expected are correct.

Reply:
Okay, I updated the test with the columns that we know should be correct and added a comment about when the answers might change.